src/Lykke.Service.BlockchainWallets/Workflow/CommandHandlers/EndBalanceMonitoringCommandHandler.cs

```diff
@@ -1,60 +1,71 @@
-using Lykke.Service.BlockchainWallets.Core.Commands;
+using Lykke.Service.BlockchainApi.Client;
+using Lykke.Service.BlockchainWallets.Core;
+using Lykke.Service.BlockchainWallets.Core.Repositories;
 using Lykke.Service.BlockchainWallets.Core.Services;
+using Lykke.Service.BlockchainWallets.Workflow.Commands;
+
 
 namespace Lykke.Service.BlockchainWallets.Workflow.CommandHandlers
 {
     [UsedImplicitly]
     public class EndBalanceMonitoringCommandHandler
     {
         private readonly IBlockchainIntegrationService _blockchainIntegrationService;
-        private readonly ILog _log;
+        private readonly IMonitoringSubscriptionRepository _monitoringSubscriptionRepository;
 
 
         public EndBalanceMonitoringCommandHandler(
             IBlockchainIntegrationService blockchainIntegrationService,
-            ILog log)
+            IMonitoringSubscriptionRepository monitoringSubscriptionRepository)
         {
             _blockchainIntegrationService = blockchainIntegrationService;
-            _log = log;
+            _monitoringSubscriptionRepository = monitoringSubscriptionRepository;
         }
 
         [UsedImplicitly]
         public async Task<CommandHandlingResult> Handle(EndBalanceMonitoringCommand command, IEventPublisher publisher)
         {
-            _log.WriteInfo(nameof(EndBalanceMonitoringCommand), command, "");
+            var apiClient = _blockchainIntegrationService.TryGetApiClient(command.BlockchainType);
 
-            try
+            if (apiClient == null)
             {
-                var apiClient = _blockchainIntegrationService.TryGetApiClient(command.IntegrationLayerId);
-
-                if (apiClient != null)
-                {
-                    await apiClient.StopBalanceObservationAsync(command.Address);
+                throw new NotSupportedException($"Blockchain type [{command.BlockchainType}] is not supported.");
+            }
+            
+            
+            const MonitoringSubscriptionType subscriptionType = MonitoringSubscriptionType.Balance;
 
-                    publisher.PublishEvent(new WalletDeletedEvent
-                    {
-                        Address = command.Address,
-                        AssetId = command.AssetId,
-                        IntegrationLayerId = command.IntegrationLayerId
-                    });
+            var address = command.Address;
+            var assetId = command.AssetId;
+            var blockchainType = command.BlockchainType;
 
-                    return CommandHandlingResult.Ok();
+            
+            await _monitoringSubscriptionRepository.UnregisterWalletSubscriptionAsync
+            (
+                blockchainType: blockchainType,
+                address: address,
+                assetId: assetId,
+                subscriptionType: subscriptionType
+            );
+            
+            if (await _monitoringSubscriptionRepository.WalletSubscriptionsCount(blockchainType, address, subscriptionType) == 0)
```

Comment:
Comment from previous revision is still actual - looks like count of subscribtions should be > 0 to stop something

Reply:
As we've discussed - this is count of the address+asset subscriptions in the storage, but in blockchain API we have just address subscription, so for multi asset blockchains we need to call API only when there are no address+asset subscriptions